subsys/console/line_fifo.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2018 Linaro Limited.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Legacy fifo-based line input
+ */
+
+#include <zephyr.h>
+#include <console.h>
+
+#ifdef CONFIG_UART_CONSOLE
+#include <drivers/console/uart_console.h>
```

Comment:
argh, this is really gross.
Other subsystems define an API struct which would have a function pointer for "register_input"
Ifdefing per console driver like this doesn't scale.
I guess we can fix this in another patch.

Reply:
agree, the registration needs to happen from the console driver, not here, lets fix this in another PR